### PR TITLE
Recent station feature

### PR DESCRIPTION
**Pull request by NguyenAnhQuoc1994**

### CODE REVIEW

File: src/main/java/com/mgmtp/radio/service/user/RecentStationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        //Get StationDTO by list recent station id
+        List<String> recentStationIdList = recentStationDTOFlux.map(RecentStationDTO::getStationId).toStream().collect(Collectors.toList());
+        Flux<StationDTO> stationDTOFlux = stationService.getListStationByListStationId(recentStationIdList);
+
+        //Sort list StationDTO
+        List<StationDTO> StationDTOListSorted = new ArrayList<>();
+        recentStationIdList.stream().forEach(recentStationId -> {
+            StationDTO stationDTOSorted = stationDTOFlux.toStream().filter(stationDTO ->
+                    stationDTO.getId().equals(recentStationId)
+            ).findFirst().get();
```
Comment (Tran-Hoang-Lam) [inline src/main/java/com/mgmtp/radio/service/user/RecentStationServiceImpl.java]: use get() without check present?

File: src/main/java/com/mgmtp/radio/service/user/RecentStationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public Flux<StationDTO> getRecentStation(String userId) {
+        Flux<RecentStationDTO> recentStationDTOFlux =
+                recentStationRepository.findByUserIdOrderByJoinedTimeDesc(userId).map(recentStationMapper::recentStationToRecentStationDTO);
+
+        //Get StationDTO by list recent station id
+        List<String> recentStationIdList = recentStationDTOFlux.map(RecentStationDTO::getStationId).toStream().collect(Collectors.toList());
+        Flux<StationDTO> stationDTOFlux = stationService.getListStationByListStationId(recentStationIdList);
+
+        //Sort list StationDTO
+        List<StationDTO> StationDTOListSorted = new ArrayList<>();
```
Comment (Tran-Hoang-Lam) [inline src/main/java/com/mgmtp/radio/service/user/RecentStationServiceImpl.java]: change StationDTOListSorted to stationDTOListSorted

File: src/main/java/com/mgmtp/radio/domain/user/RecentStation.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import java.time.LocalDateTime;
+
+@Data
+@NoArgsConstructor
+@Document(collection = "recent_station")
+public class RecentStation {
+    @Id
+    private String id;
+
+    @Indexed
```
Comment (Tran-Hoang-Lam) [inline src/main/java/com/mgmtp/radio/domain/user/RecentStation.java]: Change to @Indexed(unique = true)

**Comment by NguyenAnhQuoc1994** [comment]
Done!

**Pull request merged**